sematic/db/models/factories.py

```diff
@@ -171,7 +189,17 @@ def make_artifact(value: Any, type_: Any) -> Tuple[Artifact, bytes]:
 
     payload = json.dumps(value_serialization, sort_keys=True).encode("utf-8")
 
-    return artifact, payload
+    upload_payloads = [
+        UploadPayload(
+            namespace=StorageNamespace.artifacts, key=artifact_id, payload=payload
+        )
+    ]
+    for key, blob in blobs.items():
+        upload_payloads.append(
+            UploadPayload(namespace=StorageNamespace.blobs, key=key, payload=blob)
+        )
+
+    return artifact, tuple(upload_payloads)
 
 
 def deserialize_artifact_value(artifact: Artifact, payload: bytes) -> Any:
```

Comment:
The `"blob"` parts of `Artifact`s are only uploaded and never downloaded. Don't the values behind the Artifact require their specific binary payloads to be reconstituted?

Reply:
They are downloaded by the front-end.

This does not change the way that we store serialized artifacts. An artifact's serialization represents the entire artifact and can be deserialized at runtime to feed as input to a function.

Blobs represent fragments of artifact summaries that are too large to include in a JSON payload.

Take for example the following class:
```python
@dataclass
class Output:
    foo: Image
    bar: List[Image]
```

the artifact serialization will be a pickle of the `Output` instance.
The JSON summary exists to tell the dashboard how to render the artifact. In order to avoid saving all the images (potentially many) in the JSON payload that goes in the database, we store them as binary blobs on S3 and only include a reference to the blob in the JSON summary. When the dashboard encounters the blob reference, it will query the API to get the actual binary content.